src/planner/storage_plan.py

```diff
@@ -24,20 +25,25 @@ class StoragePlan(AbstractPlan):
 
     Arguments:
         video (DataFrameMetadata): Required meta-data for fetching data
-        batch_size (int): size of batch frame
+        batch_mem_size (int): memory size of the batch retreived
         skip_frames (int): skip frequency
         offset (int): storage offset for retrieving data
         limit (int): limit on data records to be retrieved
         total_shards (int): number of shards of data (if sharded)
         curr_shard (int): current curr_shard if data is sharded
     """
 
-    def __init__(self, video: DataFrameMetadata, batch_size: int = 1,
+    def __init__(self, video: DataFrameMetadata, batch_mem_size: int = None,
                  skip_frames: int = 0, offset: int = None, limit: int = None,
                  total_shards: int = 0, curr_shard: int = 0):
         super().__init__(PlanOprType.STORAGE_PLAN)
         self._video = video
-        self._batch_size = batch_size
+        self._batch_mem_size = batch_mem_size
+        if self._batch_mem_size is None:
+            self._batch_mem_size = ConfigurationManager(
+            ).get_value("executor", "batch_mem_size")
+            if self._batch_mem_size is None:
+                self._batch_mem_size = 300000000  # 300mb
```

Comment:
Any reason we need these logics, instead of a default value of 300mb, like what we did for the batch size in the old implementation.

Reply:
I think it was similar

Reply:
Ohh wait I changed 300mb to 30mb bcz it didn't pass on circleci. Is that what you are referring to?

Reply:
I meant we can do
```python
def __init__(self, video: DataFrameMetadata, batch_mem_size: int = 300000000,
                 skip_frames: int = 0, offset: int = None, limit: int = None,
                 total_shards: int = 0, curr_shard: int = 0):
```
Instead of the line 42-46.

Reply:
Ah! makes sense.

Reply:
I feel it's better that when we create the plan in the optimizer (i.e., rules), we choose the `batch_mem_size`. This entitles optimizer more capability. How do you think?

Reply:
Interesting. Let me think about it.

Reply:
Done